js/ui/codeView.js

```diff
@@ -0,0 +1,1623 @@
+        src.ease_property(
+            'rotation_angle_y',
+            direction == Gtk.DirectionType.RIGHT ? 90 : -90,
+            {
+                duration: WINDOW_ANIMATION_TIME * 2,
+                mode: Clutter.AnimationMode.EASE_IN_QUAD,
+                onComplete: this._rotateOutToMidpointCompleted.bind(this, src, direction),
+            });
+
+        let dst = oldDst ? oldDst : newDst;
+        dst.ease_property(
+            'rotation_angle_y',
+            direction == Gtk.DirectionType.RIGHT ? -90 : 90,
+            {
+                duration: WINDOW_ANIMATION_TIME * 2,
+                mode: Clutter.AnimationMode.EASE_IN_QUAD,
+                onComplete: this._rotateInToMidpointCompleted.bind(this, oldDst, newDst, direction),
+            });
+    }
+
+    _rotateOutToMidpointCompleted(src, direction) {
+        this._activateAppFlip();
+
+        this._setEffectsEnabled(src, true);
+
+        src.ease_property(
+            'rotation_angle_y',
+            direction == Gtk.DirectionType.RIGHT ? 180 : -180,
+            {
+                duration: WINDOW_ANIMATION_TIME * 2,
+                mode: Clutter.AnimationMode.EASE_OUT_QUAD,
+                onComplete: this._rotateOutCompleted.bind(this, false),
+            });
+    }
+
+    _rotateInToMidpointCompleted(oldDst, newDst, direction) {
+        if (oldDst) {
+            newDst.rotation_angle_y = oldDst.rotation_angle_y;
+            global.window_manager.completed_destroy(oldDst);
+        }
+
+        _setDimmedState(this._button, false);
+        this._setEffectsEnabled(newDst, false);
+
+        // Now show the destination
+        newDst.meta_window.activate(global.get_current_time());
+        newDst.opacity = 255;
+
+        newDst.ease_property('rotation_angle_y', 0, {
+                duration: WINDOW_ANIMATION_TIME * 2,
+                mode: Clutter.AnimationMode.EASE_OUT_QUAD,
+                onComplete: () => {
+                    this._restoreButtonState();
+                    this._rotateInCompleted();
+                }
+            });
+    }
+
+    // We need to keep these separate here so that they can be called
+    // by killEffects later if required.
+    _rotateInCompleted() {
+        let actor = this._rotatingInActor;
+        if (!actor)
+            return;
+
+        actor.rotation_angle_y = 0;
+        actor.opacity = 255;
+        this._rotatingInActor = null;
+    }
+
+    _rotateOutCompleted(resetRotation) {
+        let actor = this._rotatingOutActor;
+        if (!actor)
+            return;
+
+        if (resetRotation)
+            actor.rotation_angle_y = 0;
+        this._rotatingOutActor = null;
+    }
+
+    killEffects() {
+        this._rotateInCompleted();
+        this._rotateOutCompleted(true);
+    }
+});
+
+const SessionLookupFlags = {
+    SESSION_LOOKUP_APP: 1 << 0,
+    SESSION_LOOKUP_TOOLBOX: 1 << 1,
+};
+
+
+var CodeViewManager = GObject.registerClass({
+    Name: 'CodeViewManager',
```

Comment:
nitpick: The `Name` parameter isn't necessary with `GObject.registerClass()`, maybe this is a rebase leftover from `Lang.Class`?